### ISSUE TITLE
Add a Markdown report format (for use with GitHub Job Summaries)

### ISSUE THREAD

**Issue by sschuberth** [detekt/detekt#4826]
## Expected Behavior

It would be nice if there was an additional report in Markdown format (maybe even replacing the plain text format) that could be used with GitHub's new [Job Summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).

## Current Behavior

There is no report in Markdown format available.

## Context

While the SARIF report also works nicely with GitHub PRs, [Job Summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/) offer a yet nicer way to get a quick overview of all issues found. The existing plain text format does not make use of Markdown's formatting and linking capabilities, so having a dedicated Markdown report format would be nice.

**Comment by cortinico** [comment]
> (maybe even replacing the plain text format)

Sounds like a good idea, unless we know of any user currently relying on the current text format (which I'm not aware of).

**Comment by BraisGabin** [comment]
:) sure! I assigned it to you.

**Comment by schalkms** [comment]
I would also vote for this suggestion.
The plain text report doesn't provide any additional value, if detekt includes a markdown based report.

> maybe even replacing the plain text format

**Comment by BraisGabin** [comment]
I agree aboit to remove txt but for 2.0

Remove it now could break some setups.

**Comment by cortinico** [comment]
> could break some setups.

Which one? If a user is expecting structured data from a report they would be using XML or other format no?

**Comment by BraisGabin** [comment]
No idea. It just feels like a breaking change that doesn't give us too much, but maybe I'm losing something here.

**Comment by cortinico** [comment]
I feel that we don't have a strong API around the text format, given that it's fully unstructured. Adding some markdown won't hurt IMHO. If a user was relying on the text output to feed it into another tool, they should have used a more structured format maybe?

**Comment by VitalyVPinchuk** [comment]
Hello!
If I get it right, Markdown report should return string similar to this.
If no, please, correct me.

Structure:

---

1. **TestSmell-1**

    1. <details>
       <summary>CodeSmell-1</summary>

       - <details>
           <summary>Issue</summary>
         
           - id='TestSmell'
           - severity=CodeSmell
           - debt=5min
         </details>
         
       - <details>
           <summary>Entity</summary>

           - name=TestEntity
           - signature=TestEntitySignature
           - <details>
               <summary>location</summary>
          
               - source=1:1
               - text=0:0
               - file=TestFile.kt
               - <details>
                   <summary>filePath</summary>
              
                   - absolutePath=TestFile.kt
                   - basePath=null
                   - relativePath=null
                 </details>

             </details>
           - ktElement=null
         </details>

       - Message
       - metrics
       - reference
       - severity=WARNING
       - id='TestSmell'
       </details>

    1. <details>
       <summary>CodeSmell-2</summary>

       - Issue
       - <details>
           <summary>Entity</summary>

           - name=TestEntity
           - signature=TestEntitySignature
           - <details>
               <summary>location</summary>

               - source=1:1
               - text=0:0
               - file=TestFile.kt
               - <details>
                   <summary>filePath</summary>

                   - absolutePath=TestFile.kt
                   - basePath=null
                   - relativePath=null
                 </details>

             </details>
           - ktElement=null
         </details>

       - Message
       - metrics
       - reference
       - severity=WARNING
       - id='TestSmell'
     </details>
 
1.  **TestSmell-2**

---


<details>
  <summary>Markup:</summary>

```html
1. **TestSmell-1**

    1. <details>
       <summary>CodeSmell-1</summary>

       - <details>
           <summary>Issue</summary>
         
           - id='TestSmell'
           - severity=CodeSmell
           - debt=5min
         </details>
         
       - <details>
           <summary>Entity</summary>

           - name=TestEntity
           - signature=TestEntitySignature
           - <details>
               <summary>location</summary>
          
               - source=1:1
               - text=0:0
               - file=TestFile.kt
               - <details>
                   <summary>filePath</summary>
              
                   - absolutePath=TestFile.kt
                   - basePath=null
                   - relativePath=null
                 </details>

             </details>
           - ktElement=null
         </details>

       - Message
       - metrics
       - reference
       - severity=WARNING
       - id='TestSmell'
       </details>

    1. <details>
       <summary>CodeSmell-2</summary>

       - Issue
       - <details>
           <summary>Entity</summary>

           - name=TestEntity
           - signature=TestEntitySignature
           - <details>
               <summary>location</summary>

               - source=1:1
               - text=0:0
               - file=TestFile.kt
               - <details>
                   <summary>filePath</summary>

                   - absolutePath=TestFile.kt
                   - basePath=null
                   - relativePath=null
                 </details>

             </details>
           - ktElement=null
         </details>

       - Message
       - metrics
       - reference
       - severity=WARNING
       - id='TestSmell'
     </details>
 
1.  **TestSmell-2**

---
```
</details>

**Comment by cortinico** [comment]
> If no, please, correct me.

If possible, I would refrain from using `<summary>` and `<details>` as those are not standard Markdown. Something simpler also works

**Comment by BraisGabin** [comment]
I think that something similar as the `html` report (without code snippets) but using *pure* markdown could be a good start.

**Comment by VitalyVPinchuk** [comment]
<details><summary>Will such a representation suffice?</summary>


# detekt
## Metrics
   - 10,000 M1
   - 2 M2
## Complexity
   - 2,222 lines of code (loc)
   - 20 source lines of code (sloc)
   - 10 logical lines of code (lloc)
   - 2 comment lines of code (cloc)
   - 10 cyclomatic complexity (mcc)
   - 10 cognitive complexity
   - 4 number of total code smells
   - 10% comment source ratio
   - 1,000 mcc per 1,000 lloc
   - 400 code smells per 1,000 lloc
## Findings
   ### Total: 4
   1. TestSmell
      TestSmell: 1 
      Description TestSmell
      [Documentation](https://detekt.dev/docs/rules/testsmell#testsmell)
      - TestFile.kt:1:1
        ```
        TestMessage
        ```
   1. TestSmellA
      TestSmellA: 2 
      Description TestSmellA
      [Documentation](https://detekt.dev/docs/rules/testsmella#testsmella)
      - TestFile.kt:1:1
        ```
        TestMessage
        ```
      - TestFile.kt:1:1
        ```
        TestMessage
        ```
   1. TestSmellC
      TestSmellC: 1 
      Description TestSmellC
      [Documentation](https://detekt.dev/docs/rules/testsmellc#testsmellc)
      - TestFile.kt:1:1
        ```
        TestMessage
        ```
 
generated with [detekt version 1.20.0](https://detekt.dev) on 2022-05-22 21:00:19 UTC

</details>

**Comment by BraisGabin** [comment]
This was implemented so closing :)
